sdk/lib/core/uri.dart

```diff
@@ -459,21 +499,23 @@ abstract interface class Uri {
   /// The returned scheme is canonicalized to lowercase letters.
   String get scheme;
 
-  /// The authority component.
+  /// The authority (hostname) component.
   ///
-  /// The authority is formatted from the [userInfo], [host] and [port]
-  /// parts.
+  /// The authority is the [hostname] + [port]. Optionally includes
```

Comment:
The first sentence is technically incorrect. I prefer to not say anything that is technically incorrect, even if it is corrected in the very next sentence, because not all readers will read the next sentence.

So, something like:
```dart
/// The authority always incldes the [host], which can be either a host name,
/// an IPv4 address or an IPv6 address. The host optionally be followed by
/// a [port] number, and preceded by [userInfo], for example:
/// * `//example.com`
/// * `//username:password@example.com:8080`
/// * `//127.0.0.1:80` (IPv4 loopback)
/// * `//[::1]:80/` (IPv6 loopback)
```

I use the `//` before every example, because it is part of the authority.